src/lib/components/mods/ModForm.svelte

```diff
@@ -20,12 +23,34 @@
     name: '',
     short_description: '',
     source_url: '',
-    hidden: false
+    hidden: false,
+    tagIDs: [],
+    compatibility: {
+      EA: {
+        state: CompatibilityState.Works,
+        note: ''
+      },
+      EXP: {
+        state: CompatibilityState.Works,
+        note: ''
+      }
+    }
   };
   export let submitText = 'Create';
 
   export let editing = false;
 
+  let tags = [];
+  $: {
+    const anyData = $data;
+    if (anyData.tags) {
+      tags = anyData.tags;
+      delete anyData.tags;
```

Comment:
Why are you deleting the property here?

Reply:
Because the GQL Query Model for a Mod returns "tags" consisting of Tag Models while the UpdateMod Mutation Model expects a tagIDs field consisting of the Tag ID's directly.
This mutation model is also verified using Zod etc.
For this reason, this code removes the tags field after data from the query has been recieved, and replaces it with a tagIDs field.
The "tags" variable in the component is used for the TagList Edit as it operates on Tag Objects and not just the tag ids.